src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyOfState = { ...state };
+  const answerArray = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        copyOfState = Object.assign(copyOfState, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete copyOfState[key];
+        }
+        break;
+
+      case 'clear':
+        for (const item in copyOfState) {
+          delete copyOfState[item];
+        }
```

Comment:
```suggestion
        for (const item in copyOfState) {
          delete copyOfState[item];
        }
```
```suggestion
        copyOfState = {}
```
You can just make it an empty object